src/components/Footer.jsx

```diff
@@ -80,16 +79,18 @@ class Footer extends React.Component {
     sendTrackEvent(eventName, properties);
   }
 
-  CCPADialogCloseHandler() {
-    this.setState({
-      showCCPADialog: false,
-    });
-  }
-
-  CCPADialogOpen() {
-    this.setState({
-      showCCPADialog: true,
-    });
+  toggleConsentModal() {
+    window.OneTrust?.ToggleInfoDisplay();
+    const { host } = new URL(`${MARKETING_BASE_URL}`);
+    // Use global domain (`.edx.org`) without the subdomain
+    const hostParts = host.split('.');
+    const domain = hostParts.length > 2
+      ? `.${hostParts.slice(-2).join('.')}`
+      : host;
+    const cookieOptions = host.startsWith('localhost')
+      ? {}
+      : { domain, expires: 365 };
+    Cookies.set(CCPA_COOKIE_NAME, true, cookieOptions);
```

Comment:
Is this a cookie, that is meant to be immediately set to `true` just by clicking on `Your Personal Information`? This wouldn't be set if the user clicked on a toggle _within_ the modal?

Reply:
Yes, that is the correct behaviour. It's a bit strange but it was a requirement given by Legal.